### PR TITLE
Fix golint failure - "cmd/cloud-controller-manager/app/apis/config/v1alpha1" - replace PR #91869

### PR DESCRIPTION
**Pull request by jiefeng-xu**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Fix golint failure - "cmd/cloud-controller-manager/app/apis/config/v1alpha1"

**Which issue(s) this PR fixes**:

Fixes #68026

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by stantonxu** [comment]
@jiahuif I've been struggling to past the test, especially on the BUILD file. Could you please share some suggestions how to pass the errors?

File: cmd/cloud-controller-manager/app/apis/config/v1alpha1/types.go [hunk 1]
```diff
@@ -1,39 +0,0 @@
...
-package v1alpha1
-
-import (
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	kubectrlmgrconfigv1alpha1 "k8s.io/kube-controller-manager/config/v1alpha1"
-)
-
-// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
-
-type CloudControllerManagerConfiguration struct {
```
Comment (cheftako) [inline cmd/cloud-controller-manager/app/apis/config/v1alpha1/types.go:26]: This struct is used in https://github.com/kubernetes/kubernetes/blob/master/cmd/cloud-controller-manager/app/options/options.go#L118. As such you cannot expect this to build if you remove this definition.

Comment (stantonxu) [reply cmd/cloud-controller-manager/app/apis/config/v1alpha1/types.go:26]: Fixed the options.go reference. Thanks @cheftako

**Comment by stantonxu** [comment]
@cheftako , I updated the options.go file, to import `"k8s.io/kubernetes/cmd/cloud-controller-manager/app/apis/config"` instead of `"k8s.io/kubernetes/cmd/cloud-controller-manager/app/apis/config/v1alpha1"` for the `CloudControllerManagerConfiguration` struct. However, now it causes options_test.go failure. I think I need some help here.

**Comment by stantonxu** [comment]
Thanks to @cheftako 's help, by which now the number of failed jobs is down to 3.

@dims , @cheftako , I looked into the bazel test failure. It is because the `NewCloudControllerManagerOptions` doesn't come with default values. While I look back to the `v1alpha1` code I deleted for the PR, there were `defaults.go` and `defaults_test.go` files in that folder, but not corresponding code in the `apis/config` folder. Should I move them into the new folder instead of deleting them? 

And the `zz_generated.defaults.go` file was generated by `defaulter-gen` in the `v1alpha1` folder. How can I generate it into the `apis/config` folder after I delete the `v1alpha1` folder?

**Comment by stantonxu** [comment]
@dims , per my further discussion with @cheftako , it is not clear that removing the CCM app/apis/config/v1alpha1 is the correct way of proceeding. app/apis/config is the internal version of the config resource app/apis/config/v1alpha1 is the external version of the resource. We need external versions of resources so that the configs can change and be backward compatible. The intent is that an old config file should still be readable for some time after a new schema (Eg v1beta1 and then v1) has been created.

/assign @dims

**Comment by cheftako** [comment]
@stantonxu Would you like to close this?

**Pull request closed without merging**